### ISSUE TITLE
demo dont work

### ISSUE THREAD

**Issue by qinyang1018** [Karumi/Dexter#164]
### Expected behaviour
![image](https://user-images.githubusercontent.com/15726024/27897731-a09122c0-6254-11e7-8644-cb724dcf381e.png)
![image](https://user-images.githubusercontent.com/15726024/27897763-ce2f9ab8-6254-11e7-95f1-bddc5a96afa5.png)

Run demo for all permissions, no dialog show, all permissions to become granted
![image](https://user-images.githubusercontent.com/15726024/27897790-f7e46eb0-6254-11e7-9797-014cb7161a49.png)
![image](https://user-images.githubusercontent.com/15726024/27897815-207d9720-6255-11e7-8ae7-a86ca78c2362.png)

But the application without the current access permissions
### Version of the library
github demo

**Comment by Serchinastico** [comment]
Hi @qinyang1018 

I just tried in an emulator with Android 6.0 and it works fine. Have you tried completely deleting the demo app and installing it again? 

If I had to do a wild guess I'd say that the problem might be in the Smartisan OS it looks like you are using, maybe that ROM is somehow ignoring request permissions and grants them by default (just like Android did before 6.0). Have you seen the permissions alert in some other app?
